server/Main.hs

```diff
@@ -7,12 +7,12 @@
 module Main (main) where
 
 import Network.Wai.Handler.Warp
-import Data.IORef
-import Data.Time.Clock
+import Data.IntMap.Strict (empty)
+import Control.Concurrent.STM.TVar (newTVarIO)
 
 import Purestone.Server.App
 
+-- | Entry point for the server executable. Creates the 2 TVars used by the 
+--   program and uses `warp` to deploy the API
```

Comment:
In terms of commenting, I'd leave haddock comments for the description that is relevant to the caller - e.g. "Entry point for the server executable." here. The explanation of the implementation details would be better off as a comment in the body (if needed)

Reply:
I think I've done this, but I may have missed one